### PR TITLE
WIFI-1615 Correct old out of range default backup channel number sett…

### PR DESCRIPTION
**Pull request by lynnshi-design**
…ing (154) on 5GU

### CODE REVIEW

File: equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java [hunk 1]
```diff
@@ -275,19 +275,37 @@ private void validateChannelNum(Equipment equipment) {
...
 					}
 				}
 			}
 		}
 	}
 	
+	// Correct old out of range default backup channel number setting (154) on 5GU
+	private Integer correctOldInvalidBackupChannel(RadioType radioType, Integer channelNum,
+			ElementRadioConfiguration elementRadioConfig, boolean manual) {
+		if (radioType == RadioType.is5GHzU && channelNum != null && channelNum == 154) {
```
Comment (dmitry-toptygin-connectus-ai) [inline equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java:298]: So, whoever configures the channel 154 would actually receive configuration for 157 ?
Is this correct behaviour for all country codes? or only for us/canada ?
Is this permanent change or a transitionary? 
If it is transitionary - we should immediately schedule a JIRA for the future to remove the temporary code.
If that is a permanent solution, we should at least include a big warning in the log whenever this happens with explanations why it is happening.

File: equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java [hunk 2]
```diff
@@ -275,19 +275,37 @@ private void validateChannelNum(Equipment equipment) {
...
+			if (manual) {
+				elementRadioConfig.setManualBackupChannelNumber(157);
+			} else {
+				elementRadioConfig.setBackupChannelNumber(157);
+			}
+			return 157;
+		}
+		return channelNum;
```
Comment (mikehansen1970) [inline equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java:306]: This is wrong on a number of levels.
The available channels are reported in channelPowerLevels on a per radio band. These are the channels reported as allowed by the AP, and by design this is the correct set to use.
Sort them if you are looking for the uppermost allowed channel in that band, if that is the concern, but transparently changing it in the code, to a hardcoded value that may be wrong tomorrow, is never a good idea.

**Review by mikehansen1970 — CHANGES_REQUESTED** [review]
Please do not submit in this current state.

File: equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java (hunk 2 above)
Comment (lynnshi-design) [reply equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java:306]: @dmitry-toptygin-connectus-ai , @mikehansen1970 , I will discard this change. We will ask users in the community to delete/re-add AP if the error happens. Thank you.

**Comment by lynnshi-design** [comment]
Discard this change. We will ask users in the community to delete/re-add AP if the error happens

**Pull request closed without merging**